Pass latin typography values to psammead components

pjlee11 opened the pull request:
Resolves #1583
Resolves #1598

**Overall change:** 
Installs the new version of the following components as they now required typography values to be be passed in: 
- `psammead-headings` // which also has styling updates for the `<h2>`
- `psammead-paragraph`
- `psammead-caption`

Adds the typography script for each service config.

**Code changes:**

- Imports in the typography script for the each service config. 
- Uses react context to pass the script value from the service config to individual components.
- Passes script into the components stated above. 
- Bumps the new packages version 
- Unit tests for paragraph and caption container changed to stop using shallow snaphots as part of adding ContextProvider to the test setup.
- Headings container purposefully does not use `useContext` because it has enzyme tests which do not yet support `useContext` this will be resolved in https://github.com/bbc/simorgh/issues/1569
- Several snapshot changed from `em` to `rem` as well

**Testing note**

This PR will need a visually regression of both a `/news` article and a `/persian` article. 

For the news article the typography (line-heights and font-sizes) should remain the same as on a live article. There is one expected visual change which is that the `<h2>` should have bigger spacing above a below as per the screenshots commented below. 

For the persian articles the typography **is expected to change compared to the test equivalent.** The changes should be as per the values [here](https://github.com/bbc/psammead/blob/latest/packages/utilities/gel-foundations/src/scripts/arabic.js). To test these values are as expected we can ensure the following: 
- The `<h1>` above 1008px should have a computed value of 44px font-size and 54px line-height and  as per [groupD in canon](https://github.com/bbc/psammead/blob/latest/packages/utilities/gel-foundations/src/scripts/arabic.js#L11-L14). 
- The `<p>` above 1008px should have a computed value of  22px font-size and 34px line-height as per [groupD in bodyCopy](https://github.com/bbc/psammead/blob/latest/packages/utilities/gel-foundations/src/scripts/arabic.js#L81-L84)
- The `<figcaption>` above 1008px should have a computed value of  20px font-size and 26px line-height as per [groupD in longPrimer](https://github.com/bbc/psammead/blob/latest/packages/utilities/gel-foundations/src/scripts/arabic.js#L109-L112)

Helpful links
[News local article](http://localhost:7080/news/articles/cl55zn0w0l4o)
[New live article](https://www.bbc.co.uk/news/articles/c5ll353v7y9o)
[Persian local article](http://localhost:7080/persian/articles/cyddjz5058wo)
[Persian test article](https://www.test.bbc.co.uk/news/articles/cyddjz5058wo)

---

- [x] I have assigned myself to this PR and the corresponding issues
- [x] Tests added for new features
- [ ] Test engineer approval
- [x] I have followed [the merging checklist](https://github.com/bbc/simorgh-infrastructure/blob/latest/documentation/MERGE_PROCESS.md) and this is ready to merge.

Commit:
Install new psammead-headings version

Commit:
snapshot updates

pjlee11 commented:
24px + 32px = 56px spacing above `<h2>` when 600px and above. Below 600px is 24px + 24px = 48px.

---

## Above 600px
![image](https://user-images.githubusercontent.com/7791726/55964098-0284ed00-5c6c-11e9-8858-412eead2dba5.png)

---

## Below 600px
![image](https://user-images.githubusercontent.com/7791726/55964183-2ea06e00-5c6c-11e9-88bc-167a6bed311b.png)

sareh reviewed (dismissed):
👍 Looks great locally.

chris-hinds reviewed (dismissed):
LGTM 👍

Commit:
Merge branch 'latest' of github.com:bbc/simorgh into pull-in-new-psammead-headings

jamesbrumpton commented:
This needs revising. Under the current logic the headline has unexpectedly reduced in size. However, as confirmed with @pjlee11 , this was not meant as part of this ticket.

`Local`
![image](https://user-images.githubusercontent.com/25743793/56144872-121d7200-5f9b-11e9-86b8-a7258bfa97be.png)

`Test`
![image](https://user-images.githubusercontent.com/25743793/56144886-18abe980-5f9b-11e9-940a-6ba78ac27de6.png)

Commit:
Merge branch 'latest' of github.com:bbc/simorgh into pull-in-new-psammead-headings

Commit:
new versions from psammead

Commit:
service config with script value

Commit:
add script to service configs

Commit:
pass script value to paragraph, caption and headings components

Commit:
update snapshots (mainly em -> rem)

Commit:
npm audit fix

dr3 commented on src/app/containers/Headings/index.jsx:
```diff
@@ -38,7 +39,13 @@ const HeadingsContainer = ({ blocks, type }) => {
   return (
     <GridConstrain>
-      <Heading text={text}>{renderText()}</Heading>
+      <ServiceContextConsumer>
```
Just being nosey, but you could use the `useContext` hook here, might be nicer :)

pjlee11 replied:
Nice thought, just giving the hooks doc a quick glance over and
>The current context value is determined by the value prop of the nearest <MyContext.Provider> above the calling component in the tree.  
[source](https://reactjs.org/docs/hooks-reference.html#usecontext)

Worries me slightly because is someone reordered or added a new context inside the `ServiceContext` it wouldn't be obvious why their local setup is failing without prior knowledge of both the current usage of `useContext` and how `useContext` actually works.

Happy to use it if other reviewers agree with @dr3 😄 And if we do I'll document the decision somewhere

pjlee11 replied:
Ahhhh actual I've misread the docs, ignore my comment above. I hadn't realised they are describing multiple use for a the same context provider.

Commit:
use useContext syntax

Commit:
update snapshots

Commit:
caption tests with script provided

Commit:
caption -> captionContainer

Commit:
WIP: Heading tests error

Commit:
Don't use `useContext` for headings because of the enzyme test error

The headings unit tests use enzyme which is yet to support `useContext` (source: https://github.com/airbnb/enzyme/issues/2011). For now use the older syntax of nesting the component in <ServiceContextConsumer> until the issue to refactor all occurance of <ServiceContextConsumer> across the codebase - https://github.com/bbc/simorgh/issues/1569

Commit:
paragraph unit testing with context

Commit:
fix merge conflicts

pjlee11 commented:
Codeclimate complaining on a test I'm just refactoring rather than me adding the duplication

Commit:
Merge branch 'latest' into pull-in-new-psammead-headings

Commit:
package lock weirdness

Commit:
Merge branch 'latest' into pull-in-new-psammead-headings

dr3 commented on src/app/containers/Headings/index.jsx:
```diff
@@ -38,7 +39,13 @@ const HeadingsContainer = ({ blocks, type }) => {
   return (
     <GridConstrain>
-      <Heading text={text}>{renderText()}</Heading>
+      <ServiceContextConsumer>
```
Bit annoying about the enzyme issue meaning we cant use the hook here, can you make an issue to replace it/the tests please?

pjlee11 replied:
Correct, I've mentioned in the PR description this needs to be addressed.
>Headings container purposefully does not use useContext because it has enzyme tests which do not yet support useContext this will be resolved in #1569

dr3 commented on src/app/lib/config/services/thai.js:
```diff
@@ -1,8 +1,11 @@
+import { thai as thaiScript } from '@bbc/gel-foundations/scripts';
```
Curious why youve done this here for this one, but not `latin` or `arabic`

pjlee11 replied:
Because the [`const` below](https://github.com/bbc/simorgh/pull/1584/files#diff-b616aad94f6a2934cb808262980e897dR3) is called `thai` there is a naming conflict. So either we can rename the script to be more explicit EG: `thaiScript` or we could change the const declaration to be `thaiConfig` or `thaiServiceConfig`

dr3 replied:
Ohhh gotcha, fine by me :)

dr3 reviewed (commented):
LGTM, just a couple queries

dr3 reviewed (approved):
LGTM, great work

oluoluoxenfree reviewed (dismissed):
LGTM :)

dr3 reviewed (changes requested):
Yall got conflicts

Commit:
fix caption conflicts and snapshot updates

Commit:
Merge branch 'pull-in-new-psammead-headings' of github.com:bbc/simorgh into pull-in-new-psammead-headings

Commit:
Merge branch 'latest' into pull-in-new-psammead-headings

jamesbrumpton committed:
Merge branch 'latest' into pull-in-new-psammead-headings

Commit:
Merge branch 'latest' into pull-in-new-psammead-headings

Commit:
Merge branch 'latest' into pull-in-new-psammead-headings

dr3 commented:
Popping high pri on this as its blocking ws

nataliesmart commented:
@pjlee11 I've just read through the testing notes description in this ticket, and the values spec'd for the H1, paragraph, and figure caption don't match the designs. Can we confirm this? Thanks.

Commit:
Merge branch 'latest' into pull-in-new-psammead-headings

pjlee11 commented:
@nataliesmart yep happy to confirm, the values being used

pjlee11 commented:
## News local compared to live
No changes expected in the typography styling

![image](https://user-images.githubusercontent.com/7791726/56738381-64614e80-6764-11e9-97bb-1a5f0e4504e9.png)

---

## Persian local compared to live (expected changes in typography)
Expected changes in the typography styling, for the most part all font-size and line-heights have increased.

![image](https://user-images.githubusercontent.com/7791726/56738551-bf934100-6764-11e9-9220-0abfa01c108c.png)

jamesbrumpton commented:
LGTM. Happy for this to be merged
